build-paths.json

```diff
@@ -6,13 +6,17 @@
         "core_deps": [
             "bower_components/es5-shim/es5-shim.js",
             "bower_components/jquery/dist/jquery.js",
-            "bower_components/jquery-ui/ui/jquery-ui.custom.js",
+            "bower_components/jquery-ui/ui/jquery.ui.core.js",
+            "bower_components/jquery-ui/ui/jquery.ui.widget.js",
+            "bower_components/jquery-ui/ui/jquery.ui.mouse.js",
+            "bower_components/jquery-ui/ui/jquery.ui.draggable.js",
```

Comment:
TL;DR: I won't be pursing this because I'd rather have a system that handles minimizing dependencies in a more automatic fashion.  See below for details.

I was thinking that it might be better to make more tailored packages for each of the different environments.  This would make each of download smaller, but if a user is switching between multiple environments this would make the downloads big.  Part of the issue with this is that the script tags have to be executed in a particular order so we can't just have a true shared.js and then a separate environment.js because some of the shared dependencies need to be loaded later.  What I'd like to do is the JSON dependency loading system that's used for the workers, but modify it so that to use "require".  I have a prototype of this working using a custom implementation of "require" so it is possible.  Using "require" would allow us to only load the dependencies that we're actually using which would cut down our memory foot print.  Also, we could add a hook to our custom "require" implementation to trace which dependencies are used by each environment.  From this we could determine a minimal "shared.js" and a minimal "pjs_deps.js", "sql_deps.js", "web_deps.js", etc.